### DISCUSSION TITLE
RCVD_ILLEGAL_IP

### DISCUSSION THREAD

**Discussion by 0xspade** [simple-login/app#812] [Self-hosting question] [answered]
Hi,

It seems the "RCVD_ILLEGAL_IP" has already been fixed by the community, but there are no instructions for the existing installation to fix this issue. Is it possible to update the current "README.md" and also add a section on how to fix this issue since SpamAssassin does not want "240.0.0.0/24" in this IP range?

**Comment by nguyenkims** [comment, marked as answer]
Do you know what is the `RCVD_ILLEGAL_IP` error, I don't remember seeing it in SimpleLogin.

**Reply by 0xspade** [reply]
Hi @nguyenkims 

this [issue](https://github.com/simple-login/app/issues/446) and this [one](https://github.com/simple-login/app/pull/449) resolves the `RCVD_ILLEGAL_IP` error, but it does not have any proper documentation in the README file. 

Can you help me out to fix my instance? I'm not a docker wizard so needed help on this one lol.  :)

**Reply by nguyenkims** [reply]
The issues you mentioned are rather about using a different Docker network. If your self hosted instance isn't running yet, you can  re-follow the self hosting instructions but replace `240.0.0` by `10.0.0` everywhere, it should solve the `RCVD_ILLEGAL_IP` error. Btw this `RCVD_ILLEGAL_IP` error is rather a warning and doesn't matter much so you can also safely ignore it :).

**Reply by 0xspade** [reply]
I'm a little bit a perfectionist myself haha. since I have already an existing instance, I'm afraid to break it.

Can you guide me on how to do it?

BTW, Donated $100 

I hope you appreciate it.

**Reply by nguyenkims** [reply]
Thanks for your support! 

For changing the network to `10.0.0.0`, you can try with:

0) Upgrade to the latest version 3.4.0, the doc is on https://github.com/simple-login/app/blob/master/docs/upgrade.md

1) Create the new network, let's call it `sl-network2`

```
sudo docker network create -d bridge \
    --subnet=10.0.0.0/24 \
    --gateway=10.0.0.1 \
    sl-network2
```

2) Whitelist the new network in Postfix

```
postconf -e "mynetworks = 127.0.0.0/8 [::ffff:127.0.0.0]/104 [::1]/128 240.0.0.0/24 10.0.0.0/24"
```

3) Stop existing containers

```
sudo docker stop sl-email sl-migration sl-app sl-db sl-job-runner

sudo docker rm -f sl-email sl-migration sl-app sl-db
```

4) Run all containers with the new network

```
# Run the database container. Make sure to replace `myuser` and `mypassword`
docker run -d \
    --name sl-db \
    -e POSTGRES_PASSWORD=mypassword \
    -e POSTGRES_USER=myuser \
    -e POSTGRES_DB=simplelogin \
    -p 127.0.0.1:5432:5432 \
    -v $(pwd)/sl/db:/var/lib/postgresql/data \
    --restart always \
    --network="sl-network2" \
    postgres:12.1

# Run the database migration
sudo docker run --rm \
    --name sl-migration \
    -v $(pwd)/sl:/sl \
    -v $(pwd)/sl/upload:/code/static/upload \
    -v $(pwd)/dkim.key:/dkim.key \
    -v $(pwd)/dkim.pub.key:/dkim.pub.key \
    -v $(pwd)/simplelogin.env:/code/.env \
    --network="sl-network2" \
    simplelogin/app:3.4.0 flask db upgrade

# Run init data
sudo docker run --rm \
    --name sl-init \
    -v $(pwd)/sl:/sl \
    -v $(pwd)/sl/upload:/code/static/upload \
    -v $(pwd)/simplelogin.env:/code/.env \
    -v $(pwd)/dkim.key:/dkim.key \
    -v $(pwd)/dkim.pub.key:/dkim.pub.key \
    --network="sl-network2" \
    simplelogin/app:3.4.0 python init_app.py

# Run the webapp container
sudo docker run -d \
    --name sl-app \
    -v $(pwd)/sl:/sl \
    -v $(pwd)/sl/upload:/code/static/upload \
    -v $(pwd)/simplelogin.env:/code/.env \
    -v $(pwd)/dkim.key:/dkim.key \
    -v $(pwd)/dkim.pub.key:/dkim.pub.key \
    -p 127.0.0.1:7777:7777 \
    --restart always \
    --network="sl-network2" \
    simplelogin/app:3.4.0

# Run the email handler container
sudo docker run -d \
    --name sl-email \
    -v $(pwd)/sl:/sl \
    -v $(pwd)/sl/upload:/code/static/upload \
    -v $(pwd)/simplelogin.env:/code/.env \
    -v $(pwd)/dkim.key:/dkim.key \
    -v $(pwd)/dkim.pub.key:/dkim.pub.key \
    -p 127.0.0.1:20381:20381 \
    --restart always \
    --network="sl-network2" \
    simplelogin/app:3.4.0 python email_handler.py
    
# Run the job runner
docker run -d \
    --name sl-job-runner \
    -v $(pwd)/sl:/sl \
    -v $(pwd)/sl/upload:/code/static/upload \
    -v $(pwd)/simplelogin.env:/code/.env \
    -v $(pwd)/dkim.key:/dkim.key \
    -v $(pwd)/dkim.pub.key:/dkim.pub.key \
    --restart always \
    --network="sl-network2" \
    simplelogin/app:3.4.0 python job_runner.py
```

**Reply by 0xspade** [reply]
Thanks @nguyenkims
